src/GRA.Data/Repository/AvatarBundleRepository.cs

```diff
@@ -21,6 +21,47 @@ public class AvatarBundleRepository
         {
         }
 
+        public async Task<List<UserLog>> UserHistoryAsync(int userId)
+        {
+            var userLogs = await _context.UserLogs
+               .AsNoTracking()
+               .Where(_ => _.UserId == userId
+                      && !_.IsDeleted)
+               .OrderBy(_ => _.CreatedAt)
+               .ProjectTo<UserLog>(_mapper.ConfigurationProvider)
+               .ToListAsync();
+
+            foreach (var userLog in userLogs)
+            {
+                if (userLog.BadgeId != null)
+                {
+                    userLog.BadgeFilename = _context.Badges
+                        .AsNoTracking()
+                        .Where(_ => _.Id == userLog.BadgeId)
+                        .SingleOrDefault()
+                        .Filename;
+                }
+            }
+
+            return userLogs;
+        }
+
+        public async Task UpdateHasBeenViewed(int userId, int bundleId)
+        {
+            var userLog = await _context.UserLogs
+            .Where(_ => _.UserId == userId && !_.IsDeleted && _.AvatarBundleId == bundleId)
+            .ToListAsync();
+
+            foreach (var bund in userLog) {
+                if (bund != null)
```

Comment:
Is this null check necessary? It doesn't seem to me like it's possible that this could be null.